tests/flake-searching.sh

```diff
@@ -0,0 +1,24 @@
+source common.sh
+
+clearStore
+
+cp ./simple.nix ./simple.builder.sh ./config.nix $TEST_HOME
+cd $TEST_HOME
+cat <<EOF > flake.nix
+{
+    outputs = a: {
+       defaultPackage.$system = import ./simple.nix;
+       packages.$system.test = import ./simple.nix;
+    };
+}
+EOF
+mkdir subdir
+cd subdir
+
+for i in "" . "$PWD" .# .#test; do
+    nix build $i || fail "flake should be found by searching up directories"
+done
+
+for i in "path:$PWD"; do
+    ! nix build $i || fail "flake should not search up directories when using 'path:'"
+done
```

Comment:
Perhaps also test that it doesn't search after `.git` is found?

The filesystem boundary check might be tricky to test because of permissions required to mount things, so maybe it's ok to leave that one out

Reply:
Agreed, that’s worth a few extra tests (esp. given that these are rather cheap)